PersonApi/V1/Infrastructure/BodyRewind.cs

```diff
@@ -0,0 +1,45 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Hackney.Core.Middleware
+{
+    // TODO - move to common nuget package
+
+    [ExcludeFromCodeCoverage]
+    public sealed class BodyRewindMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public BodyRewindMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            context.Request.EnableBuffering();
+            await _next(context).ConfigureAwait(false);
+        }
+    }
+
+    [ExcludeFromCodeCoverage]
+    public static class BodyRewindExtensions
+    {
+        /// <summary>
+        /// Enables the HttpRequset body to be access from within a controller method.
+        /// Without this the request body is rendered unaccessible by other middleware steps.
+        /// </summary>
+        /// <param name="app">An App builder</param>
+        /// <returns>An App builder</returns>
+        public static IApplicationBuilder EnableRequestBodyRewind(this IApplicationBuilder app)
```

Comment:
As mentioned further below - can't we get the body (content) from the incoming request? Not sure if it's possible, but if it is it would save us a lot of code rewrite.

Reply:
You can, but you can only use `FromBody` once in the contoller, and I have left the incoming request object in place so that all the associated validation will get automatically executed.
In order to get the raw body text as well you must make it available again because by the time it gets to the controller, the MVC pipelines have wiped it.
The controller does manually get the raw request body text as its first line, it just needs this bit of middleware for that line to actually work.